### PR TITLE
[CI] Update GPU driver uplift process

### PR DESCRIPTION
**Pull request by pvchupin**
* Ignore sycl_update_gpu_driver job change in sycl_precommit. Full pre-commit is expected to be done in PRs generated by sycl_update_gpu_driver, not the change in sycl_update_gpu_driver.
* Add driver version to commit title.
* Remove linux_staging drivers update. linux_staging versions are used
to produce "unstable" containers in SYCL nightly workflow, but in
fact "unstable" matching "latest" currently and there are no testing
processes in the CI currently which use "unstable" containers. Not sure what
the original intention was, but until we have any *separate* testing processes
for "unstable" containers it doesn't make sense to update the versions.
* Use personal access token to kick off pre-commit testing automatically in PR created by this job.

### CODE REVIEW

File: .github/workflows/sycl_update_gpu_driver.yml [hunk 1]
```diff
@@ -3,6 +3,11 @@ name: Update GPU driver
 on:
   schedule:
     - cron:  '0 3 * * 2'
+  pull_request:
+    branches:
+    - sycl
+    paths:
+    - .github/workflows/sycl_update_gpu_driver.yml
```
[the remark this answers is not in the document]
Comment (pvchupin) [reply .github/workflows/sycl_update_gpu_driver.yml:10]: This change enforce workflow when it's getting updated, basically it triggers workflow run in pre-commit. Like on the link you added. Otherwise pre-commit won't trigger this workflow at all.

Workflow itself by its purpose creates new PRs. The reason why last run (the one you referred to) was useless is that there've been no new driver version. I need to re-run pre-commit task (this workflow itself) as soon as new driver is available. That will create a completely new separate PR with driver version update, and I expect that will kick off standard pre-commit in new PR.

File: .github/workflows/sycl_update_gpu_driver.yml [hunk 2]
```diff
@@ -19,40 +24,14 @@ jobs:
       env:
         BRANCH: ci/update_gpu_driver-linux-${{ env.NEW_DRIVER_VERSION }}
         LLVMBOT_TOKEN: ${{ secrets.LLVM_MAIN_SYNC_BBSYCL_TOKEN }}
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GITHUB_TOKEN: ${{ secrets.LLVM_MAIN_SYNC_BBSYCL_TOKEN }}
```
[the remark this answers is not in the document]
Comment (pvchupin) [reply .github/workflows/sycl_update_gpu_driver.yml:27]: Not sure. Can try. GITHUB_TOKEN is a special github variable.

File: .github/workflows/sycl_update_gpu_driver.yml [hunk 3]
```diff
@@ -19,40 +24,14 @@ jobs:
...
         cd $GITHUB_WORKSPACE
         # Set fake identity to fulfil git requirements
         git config --global user.name "GitHub Actions"
         git config --global user.email "actions@github.com"
         git checkout -B $BRANCH
         git add -u
-        git commit -m "[GHA] Uplift GPU RT version for Linux CI" || exit 0   # exit if commit is empty
+        git commit -m "[GHA] Uplift GPU RT version for Linux CI" -m "Uplift GPU RT version for Linux to $NEW_DRIVER_VERSION" || exit 0   # exit if commit is empty
         git push https://$LLVMBOT_TOKEN@github.com/${{ github.repository }} ${BRANCH}
         gh pr create --head $BRANCH --title "[GHA] Uplift GPU RT version for Linux CI" --body "Uplift GPU RT version for Linux to $NEW_DRIVER_VERSION"
```
[the remark this answers is not in the document]
Comment (pvchupin) [reply .github/workflows/sycl_update_gpu_driver.yml:37]: There is explanation in PR description. Let me know if it's not very clear.

We don't use "unstable" anywhere today, except "unstable" containers build. There are 3 areas if we'd like to do a full cleanup (removal) of "unstable" images:

1. Update GPU driver (this specific PR is doing that)
2. dependencies.json
3. sycl_containers.yaml (code you referred to)

I was hesitating on making a full cleanup, made only (1) in this patch limiting patch only to GPU driver update.
As soon as (1) is done (2) and (3) will become effectively a dead code, I expect new unstable builds won't be triggered. I can remove it or can keep it, if we ever want to have some "unstable" containers in the future with some real purpose.

Note that driver version uplift process will keep working through the "latest" images.
In other words after this change:
* https://github.com/intel/llvm/pull/6543 - this will stay
* https://github.com/intel/llvm/pull/6542 - this will stop

File: .github/workflows/sycl_update_gpu_driver.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (pvchupin) [reply .github/workflows/sycl_update_gpu_driver.yml:10]: I've updated sycl_precommit job to do not trigger pre-commit testing when sycl_update_gpu_driver is changed.
However in this specific PR it is still triggered ... because I changed sycl_precommit in this PR :)

File: .github/workflows/sycl_update_gpu_driver.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pvchupin) [reply .github/workflows/sycl_update_gpu_driver.yml]: Resolved.

**Comment by pvchupin** [comment]
It's appeared that secrets are not passed in pull_request workflow trigger. That means that such types of jobs that require secrets (to push changes or create PRs) cannot be effectively tested in pre-commit. It's appeared that the only way to test it is post-commit, also can be tested in temporary branch where change is getting committed.
Ran such testing and PR has been generated at https://github.com/intel/llvm/pull/6602

This change is ready to land.

**Review by cperkinsintel — APPROVED** [review]
LGTM

**Comment by pvchupin** [comment]
@bader, if you have further comments I'll be happy to address in a follow up PRs.

**Comment by pvchupin** [comment]
Triggered job manually, works well it seems: https://github.com/intel/llvm/pull/6608

**Comment by bader** [comment]
> * Use personal access token to kick off pre-commit testing automatically in PR created by this job.

It doesn't seem to work as expected. I'm looking at [recent PR](https://github.com/intel/llvm/pull/6663) created to update GPU driver and I see that it [tests old version](https://github.com/intel/llvm/runs/8084525204?check_suite_focus=true of the driver instead of the version it tries to update to.
@pvchupin, could you take a look, please?

File: .github/workflows/sycl_update_gpu_driver.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (bader) [reply .github/workflows/sycl_update_gpu_driver.yml]: I don't agree that we should remove "unstable" containers. From time to time the regressions in GPU driver, which block updates. Having unstable container is useful for getting latest status and for those who is not affected by the GPU driver regressions.

**Comment by pvchupin** [comment]
> > * Use personal access token to kick off pre-commit testing automatically in PR created by this job.
> 
> It doesn't seem to work as expected. I'm looking at [recent PR](https://github.com/intel/llvm/pull/6663) created to update GPU driver and I see that it [tests old version](https://github.com/intel/llvm/runs/8084525204?check_suite_focus=true of the driver instead of the version it tries to update to. @pvchupin, could you take a look, please?
> 
> UPD: the same problem can be observed in #6608 mentioned your comment above.

Thanks. It's been addressed at https://github.com/intel/llvm/pull/6723
